subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -289,11 +291,14 @@ public void outputsRemovedAfterFailureToLoadFromCache() {
         }
 
         @Override
-        public void visitFileInputs(InputFilePropertyVisitor visitor) {
-            for (InputFilePropertySpec inputFileProperty : context.getTaskProperties().getInputFileProperties()) {
+        public void visitInputFileProperties(InputFilePropertyVisitor visitor) {
+            boolean requiresInputChanges = isRequiresInputChanges();
+            boolean usesIncrementalTaskInputs = requiresInputChanges && isUsesIncrementalTaskInputs();
```

Comment:
I would mention in the names here that this is about legacy. I would also mention the effect of using `IncrementalTaskInputs` being that everything is considered incremental.

Reply:
Done.